lib/bundler/cli.rb

```diff
@@ -85,6 +85,9 @@ def help(cli = nil)
       end
     end
 
+    # Ensure `bundle help --no-color` is valid
+    all_commands["help"].disable_class_options = false
```

Comment:
Hmm. This is concerning. Seems like the change was to make sure help works without the required arguments, but in fact the change disable the option parsing. I'll try to fix it in a way that we don't broke option parser in the next version.